tsdb/db.go

```diff
@@ -880,6 +888,10 @@ func (db *DB) run() {
 
 	backoff := time.Duration(0)
 
+	// interval * 3 / 2 so that OOO compaction will happen around the same time as
+	// in-order compaction with default options
+	OOOScheduledCompact := time.NewTimer(db.opts.OutOfOrderCompactInterval * 3 / 2)
```

Comment:
We need to actually align the time w.r.t. the wall clock. So in this case, the math would be like this:
```go
	OutOfOrderCompactInterval := 2 * time.Hour // The config.
	
	nowUnix := time.Now().Unix()
	oooCompactionIntvSec := int64(OutOfOrderCompactInterval / time.Second)
	// Aligned 'now'. In this example, an even hour.
	nextCompaction := (nowUnix / oooCompactionIntvSec) * oooCompactionIntvSec
	// Move the aligned 'now' to midway of the interval. In this example, the next odd hour.
	nextCompaction += oooCompactionIntvSec / 2
	if nextCompaction < nowUnix {
		nextCompaction += oooCompactionIntvSec
	}
	timeUntilNextCompaction := time.Duration(nextCompaction-nowUnix) * time.Second

	// oooScheduledCompact := time.NewTimer(timeUntilNextCompaction) // Use this now. Note the OOO->ooo.

	fmt.Println("Now", time.Now().UTC())
	fmt.Println("Waiting time", timeUntilNextCompaction)
	fmt.Println("Compaction will happen at", time.Now().Add(timeUntilNextCompaction).UTC())
```

The print statements and comments are for explaining. In the final code, you can say something like `We align the compactions to happen with in-order compaction, which happens midway between aligned intervals of time.`

Reply:
Those make a lot of sense. When I implemented the change last time, I was actually quite confused about why we needed to align the compaction time if the alignment would change during the execution of the program (e.g. backoff from failing compaction). Now I understand what you mean by aligning with the compaction. Thank you for being extra patient with me!